functions/src/modules/Users.functions.ts

```diff
@@ -0,0 +1,55 @@
+import * as functions from 'firebase-functions';
+import * as admin from 'firebase-admin';
+
+const moment = require('moment');
+const db = admin.database();
+
+/**
+ * Process registration of webform data
+ * 1. Registers user by transferring data into /users/ path
+ * 2. Check if users already exists
+ * 
+ * @function processRegistration()
+ */
+export const processRegistration = functions.database
+.ref('/webforms/registration/{registration_id}')
+.onCreate(async (snapshot, context) => {
+  const webform = snapshot.val();
+
+  const oldUser = await db.ref("/users/").orderByChild("/emailAddress")
+                  .equalTo(webform.email_address).once("value");
+
+  // If the user already exists stop the execution and log the error
+  if (oldUser.val() !== undefined) {
+    console.warn(`Warning! User already exist with email: ${webform.email_address}`)
+    return false;
+  }
+
+  // if user email doesn't exists prepare the newUser Object
+  const newUser: any = {
+    timestamp: moment.format('x'),
+    emailAddress: webform.email_address,
+    phoneNumber: webform.contact_number,
+    isAvailableOnWhatsApp: webform.i_am_available_on_whatsapp === 1,
+    languages: webform.languages.reduce((map, lang) => {
+      map[lang] = true;
+      return map;
+    }, {}),
+    location: webform.country,
+    name: webform.your_name,
+    services: (<any>Object).values(webform.seva).join(", "),
+    experience: webform.experience,
+    influencer: webform.where_did_u_hear_about_this_seva,
+    recommendedBy: webform.recommended_by,
+    roles: webform.seva.reduce((map, role) => {
+      // Get the first letters of service name
+      const code = role.split(' ').map(name => name[0].join(''));
```

Comment:
I had already pointed it out that you cannot take first letters to get a role name because the service name can be changed, and not in all cases it will work. There should be a dictionary. Like this:

Content Reporting → CR
Something Else → XX

Reply:
So you mean that in some cases role name would not be the first letters of words like this one `Sound Quality Reporting = SQR`

BTW this is the demonstration i.e result of the script:
![image](https://user-images.githubusercontent.com/10194373/51028064-435d5f00-15b4-11e9-885a-076797e2465f.png)

Reply:
Yes, I mean exactly this.